src/HasMedia/HasMediaTrait.php

```diff
@@ -282,31 +298,14 @@ public function addMediaConversion($name)
     }
 
     /**
-     * Delete the model. The extra logic isn't handled in a model event since the boot function is unreliable
-     * for currently unknown reasons.
-     *
-     * @return bool
-     */
-    public function delete()
-    {
-        if (!parent::delete()) {
-            return false;
-        }
-
-        $this->media()->get()->map(function ($media) {
-            $media->delete();
-        });
-
-        return true;
-    }
-
-    /**
      * Delete the model, but preserve all the associated media.
      *
      * @return bool
      */
     public function deletePreservingMedia()
     {
+        $this->deletePreservingMedia = true;
```

Comment:
@sebastiandedeyne Good point and you're certainly right that it isn't 100% clean. 

On the other hand I don't think there will be many users testing the return value and calling the `delete` method after they've just called `deletePreservingMedia`.

I do feel that when a `deletePreservingMedia` gets called on the object the user doesn't intend to use it any further. So that'll probably be just the last call that object will get.

So I'm ok with this code. Sounds logical?